plugin/evm/vm_test.go

```diff
@@ -4090,6 +4090,39 @@ func TestParentBeaconRootBlock(t *testing.T) {
 	}
 }
 
+func TestNoBlobsAllowed(t *testing.T) {
```

Comment:
Is there a way to test this by actually sending a blob instead of by setting the blob gas used? I know it may seem pedantic, but I think it's good to be precise by testing the specific scenario of interest and not a proxy[^1] for it. Testing by mutating gas risks a future change breaking this if, for example, someone sets blob-gas price to zero because they assume it's not used.

[^1]: I'm completely ok with the _implementation_ checking for a proxy (gas used) of blobs being sent.

Reply:
Modified how the (invalid) block is created to include a blob tx